nepalingo-web/src/hooks/StreakContext.tsx

```diff
@@ -102,6 +102,9 @@ export const StreakProvider = ({ children }: { children: ReactNode }) => {
         console.error("Error updating streak data:", error);
       }
     }
+
+    // Update the streak state by fetching from the database after an update
+    updateStreakState();
```

Comment:
Without this, streaks will still show up as 0/the previous value after the first flashcard page load until the user refreshes their page.